core/include/cubos/core/tel/tracing.hpp

```diff
@@ -0,0 +1,74 @@
+/// @file
+/// @brief Tracing macros.
+/// @ingroup core-tel
+
+#pragma once
+
+#include <chrono>
+#include <string>
+
+#include <cubos/core/tel/level.hpp>
+
+#define SPAN(a, b) SPAN_INNER(a, b)
+#define SPAN_INNER(a, b) a##b
+
+/// @addtogroup core-tel
+/// @{
+
+/// @brief Constructs a new info span.
+/// @param name Span name.
+#define CUBOS_SPAN_INFO(name) cubos::core::tel::SpanManager::begin(name, cubos::core::tel::Level::Info)
+
+/// @brief Constructs a new debug span.
+/// @param name Span name.
+#define CUBOS_SPAN_DEBUG(name) cubos::core::tel::SpanManager::begin(name, cubos::core::tel::Level::Debug)
+
+/// @brief Constructs a new trace span.
+/// @param name Span name.
+#define CUBOS_SPAN_TRACE(name) cubos::core::tel::SpanManager::begin(name, cubos::core::tel::Level::Trace)
+
+/// @}
+
+namespace cubos::core::tel
+{
+    /// @brief Represents the unique identifier and path for a span.
+    struct SpanId
+    {
+        std::size_t id;   ///< Unique identifier.
```

Comment:
```suggestion
        std::size_t id;   ///< Identifier which distinguishes between instances of spans with the same path.
```
It's not really unique - two threads will produce spans with the same ids. Both fields are needed to get uniqueness.